core/src/main/scala/kafka/log/Log.scala

```diff
@@ -369,6 +404,16 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /**
+   * Recover the given segment. It is possible we encounter a legacy segment with index offset overflow. In that case,
```

Comment:
You don't need to change this, but for the future, full scaladocs for private methods is sort of overkill. They are internal to the class, so they change more frequently than the public apis and keeping the documentation up to date can be a pain.